app/react/kubernetes/ServicesView/ServicesDatatable/index.tsx

```diff
@@ -0,0 +1,182 @@
+import { Row, TableRowProps } from 'react-table';
+import { Shuffle, Trash2 } from 'lucide-react';
+import { useStore } from 'zustand';
+import { useRouter } from '@uirouter/react';
+import clsx from 'clsx';
+
+import { useEnvironmentId } from '@/react/hooks/useEnvironmentId';
+import { Authorized, useAuthorizations } from '@/react/hooks/useUser';
+import KubernetesNamespaceHelper from '@/kubernetes/helpers/namespaceHelper';
+import { notifyError, notifySuccess } from '@/portainer/services/notifications';
+
+import { Datatable, Table, TableSettingsMenu } from '@@/datatables';
+import { confirmDelete } from '@@/modals/confirm';
+import { useSearchBarState } from '@@/datatables/SearchBar';
+import { Button } from '@@/buttons';
+import { Link } from '@@/Link';
+
+import { useMutationDeleteServices, useServices } from '../service';
+import { Service } from '../types';
+import { DefaultDatatableSettings } from '../../datatables/DefaultDatatableSettings';
+
+import { useColumns } from './columns';
+import { createStore } from './datatable-store';
+import { ServicesDatatableDescription } from './ServicesDatatableDescription';
+
+const storageKey = 'k8sServicesDatatable';
+const settingsStore = createStore(storageKey);
+
+export function ServicesDatatable() {
+  const environmentId = useEnvironmentId();
+  const servicesQuery = useServices(environmentId);
+
+  const settings = useStore(settingsStore);
+
+  const [search, setSearch] = useSearchBarState(storageKey);
+
+  const filteredServices = servicesQuery.data?.filter(
+    (service) =>
+      settings.showSystemResources ||
+      !KubernetesNamespaceHelper.isSystemNamespace(service.Namespace)
+  );
+
+  function useCheckboxes() {
```

Comment:
Can this be set as a variable or used directly in the disableSelect? 

e.g. `disableSelect={!useAuthorizations(['K8sServiceW']}`

Reply:
or set as a varibale, then used there in disableSelect?